tests/std/tests/P1223R5_ranges_alg_find_last/test.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+using P = pair<int, int>;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::find_last(borrowed<false>{}, 42)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::find_last(borrowed<true>{}, 42)), ranges::subrange<int*>>);
+
+template <class T, class U>
+constexpr void check_value(const T& found, const U& value) {
+    if constexpr (same_as<T, P>) {
+        assert(found.first == value);
+    } else {
+        assert(found.peek().first == value);
+    }
+}
+
+struct instantiator {
+    static constexpr P haystack[3] = {{0, 42}, {2, 42}, {4, 42}};

```

Comment:
The tests below exercise "found", "not found", and "empty" cases, which is good, but they don't exercise "found multiple occurrences", i.e. the cases where `find_last` differs from `find`.